.git-blame-ignore-revs

```diff
@@ -96,3 +96,179 @@ fbbc7aa38cf23d0966b1dbd0f9825797e84c2502
 f3899ce8c6d53fc077c4e0bfa2c6f1fe56166b57
 # STYLE: Initial run of cmake-format
 efa749515db840b54f7329b261d3ec4c3a575f83
+# STYLE: Prefer local initialization
+e23a4dbd1d19c3fecd221a25e6f4c5825dc21e00
+# STYLE: Variable can be made constexpr
+e1acaf5aa8d23c82594fd1a7cc9282b3c9786b84
```

Comment:
Sorry for repeating myself, but I believe that a commit that makes the initialization of variables more local (e23a4dbd1d19c3fecd221a25e6f4c5825dc21e00) or adds more constexpr (e1acaf5aa8d23c82594fd1a7cc9282b3c9786b84) might unexpectedly (probably unintendedly) cause some behavior change, and might be relevant when doing a `git blame`. 

I would rathernot ignore anything other than formatting changes (by Python black, clang-format, etc).

Reply:
That is certainly possible as Ebrahim pointed out, but is usually undesirable. We want to make the default more convenient for most people.

Reply:
Hmm maybe the issue is a liberal interpretation of what really counts as a style commit. I agree that those two commits one may well want to include in a blame. I wouldn't have called those "STYLE".

I'll wait to see how if this discussion continues but I'm open to the option renaming `.git-blame-ignore-revs` to something that wouldn't be used by default by github's blame.

Reply:
Perhaps remove those two commits from the list, under the theory that they shouldn't have been labeled as STYLE?

Reply:
> We want to make the default more convenient for most people.

Why do people use `git blame` in the first place? I typically use it to find exactly when a specific feature or bug was introduced. A bug might also be introduced, accidentally, by a STYLE commit. My use case typically involves multiple "blame prior to change XXX/blame previous version" steps, iteratively, to find the relevant code change. 

Of course, I may not be like "most people", I don't know. 🤷

Reply:
The problem is, with a mature codebase like ITK, a lot of the steps in git blame tracing can be style commits. And when something is hard, no one likes doing it. Style commits are, by definition, not likely to cause problems, but modify a lot of files, and therefore feature in blame tracing prominently. Unless they can be automatically ignored.